pkg/build/api/types.go

```diff
@@ -503,12 +526,40 @@ type BuildRequest struct {
 	// From is the reference to the ImageStreamTag that triggered the build.
 	From *kapi.ObjectReference
 
+	// Binary indicates a request to build from a binary provided to the builder
+	Binary *BinaryBuildSource
+
 	// LastVersion (optional) is the LastVersion of the BuildConfig that was used
 	// to generate the build. If the BuildConfig in the generator doesn't match, a build will
 	// not be generated.
 	LastVersion *int
 }
 
+type BinaryBuildRequestOptions struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	AsFile string
+
+	// Commit is the value identifying a specific commit
+	Commit string
+
+	// Message is the description of a specific commit
+	Message string
+
+	// AuthorName of the source control user
+	AuthorName string
+
+	// AuthorEmail of the source control user
+	AuthorEmail string
+
+	// CommitterName of the source control user
+	CommitterName string
+
+	// CommitterEmail of the source control user
+	CommitterEmail string
```

Comment:
> @deads2k fyi. i assume this has bearing on the log options changes you were thinking about making.

Yes it does.  @kargakis if you start having problems with https://github.com/openshift/origin/issues/5070, this might be their source.